packages/plugin-ext/src/hosted/browser/worker/plugin-manifest-loader.ts

```diff
@@ -0,0 +1,112 @@
+/********************************************************************************
+ * Copyright (C) 2020 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { PluginModel, PluginPackage } from '../../../common/plugin-protocol';
+import { Endpoint } from '@theia/core/lib/browser/endpoint';
+import URI from '@theia/core/lib/common/uri';
+
+const NLS_REGEX = /^%([\w\d.-]+)%$/i;
+
+function getUri(pluginModel: PluginModel, relativePath: string): URI {
+    const ownURI = new Endpoint().getRestUrl();
+    return ownURI.parent.resolve(PluginPackage.toPluginUrl(pluginModel, relativePath));
+}
+
+function readPluginFile(pluginModel: PluginModel, relativePath: string): Promise<string> {
+    return readContents(getUri(pluginModel, relativePath).toString());
+}
+
+function readContents(uri: string): Promise<string> {
+    return new Promise((resolve, reject) => {
+        const request = new XMLHttpRequest();
+
+        request.onreadystatechange = function (): void {
+            if (this.readyState === XMLHttpRequest.DONE) {
+                if (this.status === 200) {
+                    resolve(this.response);
+                } else if (this.status === 404) {
+                    reject('NotFound');
+                } else {
+                    reject(new Error('Could not fetch plugin resource'));
+                }
+            }
+        };
+
+        request.open('GET', uri, true);
+        request.send();
+    });
+}
+
+async function readPluginJson(pluginModel: PluginModel, relativePath: string): Promise<any> {
+    const content = await readPluginFile(pluginModel, relativePath);
+    return JSON.parse(content);
+}
+
+export async function loadManifest(pluginModel: PluginModel): Promise<any> {
+    const [manifest, translations] = await Promise.all([
+        readPluginJson(pluginModel, 'package.json'),
+        loadTranslations(pluginModel)
+    ]);
+    // translate vscode builtins, as they are published with a prefix.
+    const built_prefix = '@theia/vscode-builtin-';
```

Comment:
This is to support using vscode built-ins from `npm` I think? Do you still use these in Che? 
We have a [couple of fixes](https://github.com/theia-ide/vscode-builtin-extensions/blob/master/src/package-vsix.js#L50-L71), to work-around problems with specific built-ins during vsix packaging - I do not think we do the same with the `npm` version.  

There is also [some extra massaging](https://github.com/theia-ide/vscode-builtin-extensions/blob/master/src/package-vsix.js#L113-L118) done  to each built-in's `package.json`, e.g. to make NLS work